### ISSUE TITLE
Unable to add encoding slot after an specific glyph

### ISSUE THREAD

**Issue by Saeed854** [fontforge/fontforge#4459]
* Provide a general summary of the issue in the **Title** above.  
* Before you open an issue, please check if a similar issue already exists or has been closed before.

### Important
Mark with [x] to select. Leave as [ ] to unselect.

### When reporting a bug/issue:
- [x] Screenshot
![Capture](https://user-images.githubusercontent.com/65552918/95077241-0938d700-0720-11eb-8343-2eefb069ca44.PNG)
- [x] The FontForge version and the operating system you're using
- fontforge 20200314
- windows 7
- [x] The behavior you expect to see, and the actual behavior
- I want to add an encoding slot after an specific glyph but it just comes at the end.
- [x] Steps to reproduce the behavior
- select a glyph, go to : encoding --> add encoding slot. and the new encoding slot comes at the end of the font.
- [ ] \(optional) Possible solution/fix/workaround

### When you open an issue for a change/improvement/feature request:
- [x] A description of the problem you're trying to solve, including _why_ you think this is a problem
I just want to add a glyph after another glyph. for example I selected "ain" glyph. I want to add an glyph (encoding slot) after it but it always comes at the end of the glyph list.
![Capture](https://user-images.githubusercontent.com/65552918/95078104-7731ce00-0721-11eb-81c0-2fc9e19d0d51.PNG)

- [ ] If the feature changes current behavior, reasons why your solution is better

**Comment by aminabedi68** [comment]
-use a text editor(sth like notpade++) and open the source sfd file
-cut your glyph data code(whole your glyph code from `StartChar` to `EndChar`)
-paste after your desired glyph.

**Comment by KamilLanda** [comment]
@aminabedi68 - I afraid it is not so easy. There is tag Encoding: in the SFD file for each glyph, and there are 3 numberings in this tag. If your SFD file is sorted well then it looks like

![encoding1](https://user-images.githubusercontent.com/61148366/95352100-2c9b8780-08c2-11eb-9d09-e035f8f45a0f.png)

But if you have glyphs in the regular unicode positions (for example space U+0020) and also in non-unicode (value=-1) positions (U+????) like this

![uni](https://user-images.githubusercontent.com/61148366/95352122-36bd8600-08c2-11eb-9821-a6f673d91c66.png)

and then you add new glyph somewhere to the unicode range (Encoding/ Reencode/ UNI 10646-1 Unicode Full; Encoding/ Compact ), then the numberings in SFD file will change.

![encoding2](https://user-images.githubusercontent.com/61148366/95352153-4046ee00-08c2-11eb-92c4-c29d79a6b77d.png)

You can renumber it to the prettier form by "Encoding/ Add encoding slots" and then "Encoding/ Remove unused slots", but it will not repair the numberings to the optimal form.

![encoding3](https://user-images.githubusercontent.com/61148366/95352193-4a68ec80-08c2-11eb-8b15-9827c2803cff.png)

If you change only the place for some glyph in SFD file and not change the numberings, then glyph will be in the same position.

**Comment by skef** [comment]
You might or might not want to look at https://github.com/skef/fontforge_coverage_tools/blob/master/doc/ff_sfd_pseudoenc.md for reference. This is a tool I wrote to manage the display of non-encoded glyphs. It is a filter that modifies SFD files "offline".

**Comment by Saeed854** [comment]
Amir Abedi's solution: not worked.
KamilLanda's solution: it adds an slot but removes the next glyph. I don't know why it happens.
skef's solution: Well, that's very complicated. I don't even know how to work with it. How can I install it?

**Comment by aminabedi68** [comment]
i forgot about the numbers @Saeed854 ,you're right. but you can fix them manually (if you have just a few glyphs you can rewrite the numbers for the next glyphs after inserted glyph till end), it worked for me before.

**Comment by skef** [comment]
I believe there have been past issues requesting the ability to reorder unencoded glyphs. There hasn't been much impetus towards or interest in a fix. What you're getting is advice about workarounds. It's certainly a thing that's possible to control but you have to do it at the file level right now.

I wrote the tool I linked to as a way of managing the display of unencoded glyphs for fonts that have a lot of them. (I was doing some work on Libertinus at the time.) It's the sort of thing that would be handy to some people if they were already familiar with it interesting enough to make people become familiar with it. C'est la vie.

**Comment by Saeed854** [comment]
@aminabedi68 Yes. That method is okay for a few glyphs but not for a great amount of glyphs.
